### PR TITLE
#8102 Add extension console activation e2e test for mod with built-in integration

### LINKED ISSUE #8102: Add Playwright test for activating a mod with a built-in integration config
Create a Playwright test that effectively replaces the manual regression checklist test "Activate a mod that automatically uses the built-in integration configuration" under the "Extension Console Activation" test group: https://www.notion.so/pixiebrix/Release-Checklist-Template-6d435dbf9a464f119ac6202b05d9c3b5?pvs=4#3afe7c8fc2f84c70a6ab86a8b5c1bcc0

### Acceptance Criteria

Add a Playwright test that:
- Activates a mod via the Extension console that uses an integration with a built-in integration config, e.g. `@pixies/giphy/giphy-search`
- Make sure to assert that the integration field on the activation page is pre-populated with the built-in integration (the user should not have to configure anything to activate the mod)
- Verify that the mod works in https://pbx.vercel.app/
- May involve mocking the integration
- After merging, remove the test from the regression checklist notion template https://www.notion.so/pixiebrix/Release-Checklist-Template-6d435dbf9a464f119ac6202b05d9c3b5?pvs=4#3afe7c8fc2f84c70a6ab86a8b5c1bcc0

### PR DESCRIPTION
**Pull request by mnholtz**
## What does this PR do?

- Closes #8102 

## Reviewer Tips

Outstanding work
- [x] Add steps for getting around the user gesture modal for opening the sidebar
- [x] Mock the giphy api

## Checklist

- [x] Add jest or playwright tests and/or storybook stories
- [x] Designate a primary reviewer @fungairino

### CODE REVIEW

**Comment by twschiller** [comment]
FWIW, this test also uses a built-in integration: https://github.com/pixiebrix/pixiebrix-extension/blob/6e2f02e1d6f2c6fa8bf8f96be25278bc4586bbe5/end-to-end-tests/tests/smoke/sidebar.spec.ts#L35-L35

**Comment by twschiller** [comment]
> Add steps for getting around the user gesture modal for opening the sidebar

In some cases, the user gesture might not show up. The gotcha is the way the mod is built -- I think it will depend on how quickly the API responds. But I would suspect in most cases the user will get the user gesture dialog

**Comment by mnholtz** [comment]
> If you mock the giphy API, you might not need to worry about the user gesture dialog in this test because you could ensure it the mock responds instantaneously.

Fantastic tip, I'll start with that first today!

**Comment by mnholtz** [comment]
> FWIW, this test also uses a built-in integration

Welp, maybe I'll be extra thorough with my assertions at least

File: .idea/icon.svg
Comment (mnholtz) [inline .idea/icon.svg:1]: Unrelated - added a project icon to the extension for intellij

File: end-to-end-tests/tests/extensionConsoleActivation.spec.ts [hunk 1]
```diff
@@ -41,3 +44,58 @@ test("can activate a mod with no config options", async ({
...
+        service_id: "@pixies/giphy/giphy-service",
+      });
+
+      return route.fulfill({
+        path: path.join(__dirname, "../fixtures/responses/giphy-search.json"),
+      });
+    }
+
+    return route.continue();
+  });
```
Comment (mnholtz) [inline end-to-end-tests/tests/extensionConsoleActivation.spec.ts:73]: We can talk about extracting this pattern once we get a few more request mocking tests under our belt.

**Comment by mnholtz** [comment]
For some reason, the quickbar isn't appearing only in CI. Local tests are passing.

File: end-to-end-tests/tests/extensionConsoleActivation.spec.ts [hunk 2]
```diff
@@ -41,3 +44,65 @@ test("can activate a mod with no config options", async ({
...
+  const modActivationPage = new ActivateModPage(page, extensionId, modId);
+  await modActivationPage.goto();
+
+  await expect(
+    page.locator(".form-group").filter({ hasText: /^GIPHY — ✨ Built-in$/ }),
+  ).toBeVisible();
+  await modActivationPage.clickActivateAndWaitForModsPageRedirect();
+  await page.goto("/");
+
+  await page.waitForTimeout(5000);
```
Comment (twschiller) [inline end-to-end-tests/tests/extensionConsoleActivation.spec.ts:84]: Why is this timeout required?

File: end-to-end-tests/tests/extensionConsoleActivation.spec.ts [hunk 3]
```diff
@@ -41,3 +44,65 @@ test("can activate a mod with no config options", async ({
...
+  await page.getByText("Index of  /").click();
+  await page.keyboard.press("Meta+M");
+  await page.keyboard.press("Control+M");
+  await page.getByRole("option", { name: "GIPHY Search" }).click();
+
+  // Open quickbar; it may take a few seconds for the content script to be ready
```
Comment (twschiller) [inline end-to-end-tests/tests/extensionConsoleActivation.spec.ts:90]: I'd recommend we extract out the "Open Quickbar" logic, and potentially expose a method like `runFromQuickbar(page, optionName)`

**Review by twschiller — APPROVED** [review]

File: end-to-end-tests/tests/extensionConsoleActivation.spec.ts (hunk 2 above)
Comment (mnholtz) [reply end-to-end-tests/tests/extensionConsoleActivation.spec.ts:84]: I'm going to remove this soon. It was to test a hypothesis with failing tests in CI, no longer needed

File: end-to-end-tests/tests/extensionConsoleActivation.spec.ts (hunk 3 above)
Comment (mnholtz) [reply end-to-end-tests/tests/extensionConsoleActivation.spec.ts:90]: Agreed

File: src/utils/uuidUtils.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+export const VALID_UUID_REGEX =
```
Comment (twschiller) [inline src/utils/uuidUtils.ts:18]: Consider putting next to the uuid type instead

File: end-to-end-tests/tests/extensionConsoleActivation.spec.ts [hunk 5]
```diff
@@ -41,3 +44,57 @@ test("can activate a mod with no config options", async ({
...
+  extensionId,
+  context,
+}) => {
+  const modId = "@pixies/giphy/giphy-search";
+
+  // The giphy search request is proxied through the PixieBrix server, which is kicked off in the background/service
+  // worker. Playwright experimentally supports mocking service worker requests, see
+  // https://playwright.dev/docs/service-workers-experimental#routing-service-worker-requests-only
+  await context.route("https://app.pixiebrix.com/api/proxy/", async (route) => {
+    if (route.request().serviceWorker()) {
```
Comment (fungairino) [inline end-to-end-tests/tests/extensionConsoleActivation.spec.ts:59]: Nice job on the first network request mock!

File: end-to-end-tests/tests/extensionConsoleActivation.spec.ts (hunk 1 above)
Comment (fungairino) [reply end-to-end-tests/tests/extensionConsoleActivation.spec.ts:73]: Agreed! Probably moving them into fixtures would be a good idea.

File: end-to-end-tests/utils.ts [hunk 6]
```diff
@@ -81,10 +81,22 @@ export async function getSidebarPage(page: Page, extensionId: string) {
...
+
+    // In MV3, the sidebar sometimes requires the user to interact with modal to open the sidebar via a user gesture
+    const conditionallyPerformUserGesture = async () => {
+      await expect(page.getByRole("button", { name: "OK" })).toBeVisible();
+      await page.getByRole("button", { name: "OK" }).click();
+      return findSidebarPage(page);
+    };
+
+    await expect(async () => {
+      sidebarPage = await Promise.race([
```
Comment (fungairino) [inline end-to-end-tests/utils.ts:93]: nice use of Promise.race here!

**Review by fungairino — APPROVED** [review]

File: end-to-end-tests/tests/extensionConsoleActivation.spec.ts (hunk 3 above)
Comment (mnholtz) [reply end-to-end-tests/tests/extensionConsoleActivation.spec.ts]: I might hold off on extracting this for now. There is a quirk where the page needs to be focused via a `click()` in order for the keyboard shortcut to work. I don't fully understand it yet, and can't generalize it to any page before I can understand what's going on.

File: end-to-end-tests/tests/extensionConsoleActivation.spec.ts (hunk 5 above)
Comment (mnholtz) [reply end-to-end-tests/tests/extensionConsoleActivation.spec.ts:59]: Thanks eduardo!

File: end-to-end-tests/utils.ts (hunk 6 above)
Comment (mnholtz) [reply end-to-end-tests/utils.ts:102]: I think I got that idea from you?? But thanks!

**Pull request merged**